### DISCUSSION TITLE
Session is automatically restored for a root .git directory

### DISCUSSION THREAD

**Discussion by eldar** [rmagatti/auto-session#354] [Q&A] [answered]
I'd like to keep a session for my `~/dotfiles/.config/nvim` for editing my editor config, but whenever I restore the session it automatically restores the session for `~/.dotfiles`, presumably because this is top level directory of a git repository. I removed the corresponding session file for dotfiles in `~/.local/share/nvim/sessions/` but it still automatically creates a session for the git root. Is there any way to disable it?

**Comment by cameronr** [comment]
Hmm, that is a bit surprising. The logic doesn't do any directory hierarchy traversal (for git or otherwise). What directory are you starting `nvim` in (`~/dotfiles/.config/nvim` or `~/.dotfiles` or something else)?

Can you add `log_level = 'debug'` to your auto-session config and post the log (can get it via either `:messages` or `:Notifications`) when it doesn't restore as expected?

**Comment by eldar** [comment, marked as answer]
Hi, @cameronr! Thanks for the prompt response. Thanks for pointing out that the plugin doesn't do directory traversal itself - turns out was performed by another plugin I forgot I had enabled: `ahmedkhalf/project.nvim`. By default it searches for a toplevel git directory to be selected as workspace root - nothing to do with auto-session. Sorry for the trouble!
